cpp/src/unary/cast_ops.cu

```diff
@@ -199,7 +199,13 @@ std::unique_ptr<column> rescale(column_view input,
       }
       return output_column;
     }
-    auto const scalar = make_fixed_point_scalar<T>(std::pow(10, -diff), scale_type{diff}, stream);
+
+    RepType scalar_value = 10;
```

Comment:
Is the next step to consolidate this logic into a single function somewhere?

Reply:
Yes. See #14243.